executor/index_merge_reader.go

```diff
@@ -699,21 +735,29 @@ func (e *IndexMergeReaderExecutor) getResultTask() (*indexMergeTableTask, error)
 	return e.resultCurr, nil
 }
 
-func (e *IndexMergeReaderExecutor) handleHandlesFetcherPanic(ctx context.Context, resultCh chan<- *indexMergeTableTask, worker string) func(r interface{}) {
+func (e *IndexMergeReaderExecutor) handleHandlesFetcherPanic(ctx context.Context, finished <-chan struct{}, ch chan<- *indexMergeTableTask, worker string) func(r interface{}) {
 	return func(r interface{}) {
 		if r == nil {
 			return
 		}
 
-		err4Panic := errors.Errorf("panic in IndexMergeReaderExecutor %s: %v", worker, r)
+		err4Panic := errors.Errorf("IndexMergeReaderExecutor %s: %v", worker, r)
 		logutil.Logger(ctx).Error(err4Panic.Error())
 		doneCh := make(chan error, 1)
 		doneCh <- err4Panic
-		resultCh <- &indexMergeTableTask{
+		task := &indexMergeTableTask{
 			lookupTableTask: lookupTableTask{
 				doneCh: doneCh,
 			},
 		}
+		select {
+		case <-ctx.Done():
+			return
+		case <-finished:
+			return
+		case ch <- task:
+			return
+		}
```

Comment:
Looks most of the `handleXXXpanic` use the same code, can we reuse the code?

Reply:
Done. processWorker/partialTableWorker/partialIndexWorker all use `handleWorkerPanic()`. tableScanWorker uses `handleTableScanWorkerPanic()`, because it sends err by task.doneCh instead of resultCh or fetchCh.